dolphin/task_manager/manager.py

```diff
@@ -67,19 +70,38 @@ def say_hello(self, context, request_spec=None,
             pass
 
     def sync_storage_resource(self, context, storage_id, resource_task):
-        LOG.debug("Received the sync_storage task: {0} request for storage"
-                  " id:{1}".format(resource_task, storage_id))
-        cls = importutils.import_class(resource_task)
-        device_obj = cls(context, storage_id)
-        device_obj.sync()
-
+        lock = coordination.Lock(storage_id + resource_task)
+        if lock.acquire(False):
```

Comment:
lock.acquire(False) looks to be having meaning of acquire is TRUE , little confusing